Build/SaveLoadScreen.cc

```diff
@@ -1473,12 +1675,60 @@ static void FailedLoadingGameCallBack(MessageBoxReturnValue const bExitValue)
 
 void DoQuickSave()
 {
-	if (SaveGame(0, L"")) return;
+  // Use the Dead is Dead function if we are in DiD
+  if (gGameOptions.ubGameSaveMode == DIF_DEAD_IS_DEAD)
+    DoDeadIsDeadSave();
+  else
+  {
+    if (SaveGame(0, L"")) return;
+
+    if (guiPreviousOptionScreen == MAP_SCREEN)
+      DoMapMessageBox(MSG_BOX_BASIC_STYLE, zSaveLoadText[SLG_SAVE_GAME_ERROR], MAP_SCREEN, MSG_BOX_FLAG_OK, NULL);
+    else
+      DoMessageBox(MSG_BOX_BASIC_STYLE, zSaveLoadText[SLG_SAVE_GAME_ERROR], GAME_SCREEN, MSG_BOX_FLAG_OK, NULL, NULL); 
+  }
+}
 
-	if (guiPreviousOptionScreen == MAP_SCREEN)
-		DoMapMessageBox(MSG_BOX_BASIC_STYLE, zSaveLoadText[SLG_SAVE_GAME_ERROR], MAP_SCREEN, MSG_BOX_FLAG_OK, NULL);
-	else
-		DoMessageBox(MSG_BOX_BASIC_STYLE, zSaveLoadText[SLG_SAVE_GAME_ERROR], GAME_SCREEN, MSG_BOX_FLAG_OK, NULL, NULL);
+// Save function for Dead Is Dead
+void DoDeadIsDeadSave()
+{
+ // Check if we are in a sane! Do not save if:
+ // - we are in an AI Turn
+ // - we are in a Dialogue
+ // - we are in Meanwhile.....
+ // - we are currently in a message box - The Messagebox would be gone without selection after loading
+ if (gTacticalStatus.ubCurrentTeam == OUR_TEAM && !gfInTalkPanel && !gfInMeanwhile && !gfPreBattleInterfaceActive && guiPreviousOptionScreen != MSG_BOX_SCREEN) {
+			// Backup old saves
+			BackupSavedGame(gGameSettings.bLastSavedGameSlot);
+			// Save the previous option screen State to reset it after saving
+			ScreenID tmpGuiPreviousOptionScreen = guiPreviousOptionScreen;
+			// We want to save the current screen we are in. Unless we are in Options, Laptop, or others
+			guiPreviousOptionScreen = guiCurrentScreen;
```

Comment:
Thanks.
This was indeed very complex and unclean code that took me a long time to understand again.
I simplified it and hope I didn't break anything in refactoring.

Figuring out what has to happen here in the first place was the biggest headache of the entire gamemode, btw.